### DISCUSSION TITLE
Excerpt images aren't loaded in webspace if a Sulu role doesn't have the view permission. Is this working correctly?

### DISCUSSION THREAD

**Discussion by NeuralClone** [sulu/sulu#6804] [Q&A] [answered]
I recently ran into a rather odd problem when using excerpt images. If more than one Sulu role is defined in the admin panel, I set `permission-check="true"`, and one of the roles doesn't have the view permission for a media collection the images are part of, the images aren't loaded along with the rest of the excerpt and taxonomy data when viewing the page outside of the admin panel.

I've defined a security system for my webspace as follows:

```xml
<security permission-check="true">
    <system>Website</system>
</security>
```

The following example uses teasers. But the same problem occurs if you replace them with smart content using the page provider and access the excerpt images that way.

The `mediaId` ends up being `null` unless all Sulu roles have the view permission for the media collection it's being loaded from. If I load the image in any other context within the webspace, it loads without any issues. So the problem seems to be specific to excerpt images. I've been able to reproduce this consistently in clean sulu/skeleton-based projects with minimal configuration.

All Sulu roles with view permission on collection:
![Screenshot_01](https://user-images.githubusercontent.com/678221/186358077-1e1900c1-b3db-48e5-8ce5-5f8757cfce21.jpg)

One Sulu role without view permission on collection:
![Screenshot_02](https://user-images.githubusercontent.com/678221/186358091-6c1fc46b-cac8-4980-a6cd-9e928235e14f.jpg)

If, however, I try to load the image without dealing with excerpt images, it loads as expected:

![Screenshot 2022-08-24 034814](https://user-images.githubusercontent.com/678221/186361788-ee0440ab-fb36-4216-99b2-2a4a626f18b6.png)

If I get rid of `permission-check="true"` in my webspace configuration, the Sulu role permissions don't seem to be in effect and the images load as expected. That aspect of this makes sense and seems to be working as expected.

Since I was unsure if this is actually a bug or configuration problem, I thought it made sense to ask about it first. Is this the intended behavior?

If so, it seems counterintuitive since the Sulu and Website security systems shouldn't know anything about how the other operates. If this isn't behaving as it should, I'm willing to open an issue detailing the above behavior and steps to reproduce it.

Thanks in advance!

**Comment by NeuralClone** [comment]
Unfortunately, after spending more time exploring this issue over the past few days, I didn't really find a good solution.

Everything works fine if there's a single Sulu role. But it breaks on fresh projects with minimal configuration if I add additional roles and try to restrict their view access to the collection being used.

Instead of trying to continue playing with this and not getting anywhere, I've decided to remove the permission check option in my webspace configuration and change my approach. Manually checking permissions works as expected. So I'll just be doing that where needed.

**Comment by niklasnatter** [comment, marked as answer]
Hey, thanks for reporting this! I just debugged the issue and I think this is a bug indeed. I created a pull request to fix the behaviour here: https://github.com/sulu/sulu/pull/6825
Could you test the suggested changes and report back if they fixes the issue for you?

Thanks!

**Reply by NeuralClone** [reply]
Hi, thank you for taking a look at this and working on a fix.

The suggested changes fix the issue on my end. I haven't been able to reproduce the bug since applying them.
